File: sdk/java-sdk/src/main/java/com/evernym/verity/sdk/protocols/presentproof/common/ProposedPredicate.java
package com.evernym.verity.sdk.protocols.presentproof.common;

import com.evernym.verity.sdk.utils.AsJsonObject;
import org.json.JSONObject;

public class ProposedPredicate implements AsJsonObject {
    final JSONObject data;


    /**
     * Constructs the ProposedPredicate object
     * @param name the predicate name
     * @param credDefId the credential definition used for this predicate.
     * @param threshold threshold value of the predicate.
     */
    public ProposedPredicate(String name, String credDefId, int threshold){
        this.data = new JSONObject()
                .put("name", name)
                .put("cred_def_id", credDefId)
                .put("predicate", ">=")
                .put("threshold", threshold);
    }

    /**
     * Convert this object to a JSON object
     *
     * @return this object as a JSON object
     */
    @Override
    public JSONObject toJson() {
        return data;
    }
}

File: sdk/dotnet-sdk/src/VeritySDK/Protocols/PresentProof/Common/ProposedPredicate.cs
using System.Json;
using VeritySDK.Utils;

namespace VeritySDK.Protocols.PresentProof
{
    /// <summary>
    /// A holder for the ProposedPredicate object
    /// </summary>
    public class ProposedPredicate : AsJsonObject
    {
        JsonObject data;

        /// <summary>
        /// Constructs the ProposedPredicate object 
        /// </summary>
        /// <param name="name">the predicate name</param>
        /// <param name="credDefId">the credential definition used for this predicate</param>
        /// <param name="threshold">threshold value of the predicate</param>
        public ProposedPredicate(string name, string credDefId, int threshold)
        {
            this.data = new JsonObject();
            data.Add("name", name);
            data.Add("cred_def_id", credDefId);
            data.Add("predicate", ">=");
            data.Add("threshold", threshold);
        }

        /// <summary>
        /// Convert this object to a JSON object 
        /// </summary>
        /// <returns>this object as a JSON object</returns>
        public JsonObject toJson()
        {
            return data;
        }
    }
}